backend/src/hatchling/builders/sdist.py

```diff
@@ -37,6 +37,9 @@ def __init__(self, name: str, *, reproducible: bool) -> None:
 
         raw_fd, self.path = tempfile.mkstemp(suffix='.tar.gz')
         self.fd = os.fdopen(raw_fd, 'w+b')
+        file_stat = os.stat(self.path)
+        new_mode = normalize_file_permissions(file_stat.st_mode)
+        os.chmod(self.path, new_mode)
```

Comment:
Can you please abstract this into a utility called `normalize_artifact_permissions`?